contracts/modules/UpgradableModuleFactory.sol

```diff
@@ -81,10 +85,14 @@ contract UpgradableModuleFactory is ModuleFactory {
         require(moduleToSecurityToken[_module] == msg.sender, "Incorrect caller");
         // Only allow issuers to upgrade in single step verisons to preserve upgradeToAndCall semantics
         uint256 newVersion = modules[msg.sender][_module] + 1;
-        /* uint256 newVersion = moduleVersions[msg.sender] + 1; */
-        require(newVersion <= latestVersion, "Incorrect version");
-        OwnedUpgradeabilityProxy(address(uint160(_module))).upgradeToAndCall(logicContracts[newVersion].version, logicContracts[newVersion].logicContract, logicContracts[newVersion].logicData);
+        require(newVersion <= latestUpgrade, "Incorrect version");
+        OwnedUpgradeabilityProxy(address(uint160(_module))).upgradeToAndCall(logicContracts[newVersion].version, logicContracts[newVersion].logicContract, logicContracts[newVersion].upgradeData);
```

Comment:
Why do we need to cast `_module` to `address payable`?

Reply:
Agreed - not needed.

Reply:
Actually sorry - we do need to cast to avoid:
TypeError: Explicit type conversion not allowed from non-payable "address" to "contract OwnedUpgradeabilityProxy", which has a payable fallback function.